WordPress/Sniffs/PHP/DevelopmentFunctionsSniff.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restrict the use of various development functions.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.11.0
+ */
+class WordPress_Sniffs_PHP_DevelopmentFunctionsSniff extends WordPress_AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Groups of functions to restrict.
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type'      => 'error' | 'warning',
+	 * 		'message'   => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'error_log' => array(
+				'type'      => 'error',
+				'message'   => '%s() Debug code is not to be used in Production',
+				'functions' => array(
+					'error_log',
+					'var_dump',
+					'var_export',
+					'print_r',
+					'trigger_error',
+					'set_error_handler',
+					'debug_backtrace',
+					'debug_print_backtrace',
+					'wp_debug_backtrace_summary',
+				),
+			),
+
+			'prevent_path_disclosure' => array(
```

Comment:
I am fine with changing it to `warning` but I believe they should belong in the category of development functions. We could always remove it from the `extra` ruelset. I don't think every plugin needs to use these functions and they definitely should not be used in themes.

By having the development functions together it can be easily enabled or disabled.